teamcity/datadog_checks/teamcity/common.py

```diff
@@ -92,9 +92,8 @@ def construct_event(check, new_build):
     event_tags = deepcopy(check.build_tags)
     event_tags.extend(['build_id:{}'.format(build_id), 'build_number:{}'.format(build_number)])
 
-    if check.is_deployment:
+    if check.is_deployment or build_config_type == 'deployment':
         teamcity_event = deepcopy(DEPLOYMENT_EVENT)
-        teamcity_event['tags'].append('type:deployment')
```

Comment:
Removed this since this is redundant. The `type` tag already gets added to the build_tags when new builds are collected: https://github.com/DataDog/integrations-core/blob/96b954262055694a3b3b84625c52acdf1796e2b8/teamcity/datadog_checks/teamcity/teamcity_rest.py#L320